### ISSUE TITLE
We need to implement infographic templates

### ISSUE THREAD

**Issue by timlinux** [inasafe/inasafe#4254]

**Comment by timlinux** [comment]
@myarjunar this resolve #4146

**Comment by timlinux** [comment]
@myarjunar please work with @ivanbusthomi on this

**Comment by myarjunar** [comment]
see #3473

**Comment by myarjunar** [comment]
Hi @timlinux , I need your confirmation here :grimacing: @ivanbusthomi already provided the template for the infographic here https://github.com/inasafe/inasafe/issues/3473#issuecomment-318575052. And looking to the template, I think we need at least around 100 variables to fill all elements in the template. Is it alright?

**Comment by timlinux** [comment]
It seems like something is wrong if we need so many vars
we should be getting the diplacement etc numbers from the data table
have you figured out how to change the data source of a table at run time using the python api?
if not you may need to do some xml manipulation to change it manually before loading the renderer
(== ugly hack for now until @Gustry gets the data driven table source implemented in 3.0 code base)

**Comment by timlinux** [comment]
Can you provide an enumerated list of all the vars you think we need?

**Comment by timlinux** [comment]
It seems like something is wrong if we need so many var.
We should be getting the diplacement etc numbers from the data table
Have you figured out how to change the data source of a table at run time using the python api?
if not you may need to do some xml manipulation to change it manually before loading the renderer
(== ugly hack for now until @Gustry gets the data driven table source implemented in 3.0 code base)

Can you provide an enumerated list here of the vars you think we need to define?

**Comment by myarjunar** [comment]
AFAIK if we are using expression, then we can set the layer and feature source for the value that we want to show. And here is the list of variables if we want to replace all of the elements using variable (only 1 page of template):

```
[%@impact_function_name%]
[%@map_overview_header%]
[%@population_chart_header%]
[%@people_section_header%]
[%@total_affected%]
[%@total_affected_scale%]
[%@total_affected_header%]
[%@total_displaced%]
[%@total_displaced_scale%]
[%@total_displaced_header%]
[%@people_section_notes_html%]
[%@age_gender_section_header%]
[%@total_infant_displaced%]
[%@total_infant_displaced_scale%]
[%@total_infant_displaced_percentage%]
[%@total_infant_displaced_header%]
[%@total_child_displaced%]
[%@total_child_displaced_scale%]
[%@total_child_displaced_percentage%]
[%@total_child_displaced_header%]
[%@total_adult_displaced%]
[%@total_adult_displaced_scale%]
[%@total_adult_displaced_percentage%]
[%@total_adult_displaced_header%]
[%@total_elderly_displaced%]
[%@total_elderly_displaced_scale%]
[%@total_elderly_displaced_percentage%]
[%@total_elderly_displaced_header%]
[%@total_male_displaced%]
[%@total_male_displaced_scale%]
[%@total_male_displaced_percentage%]
[%@total_male_displaced_header%]
[%@total_female_displaced%]
[%@total_female_displaced_scale%]
[%@total_female_displaced_percentage%]
[%@total_female_displaced_header%]
[%@age_gender_section_notes%]
[%@vulnerability_section_header%]
[%@total_under_5_displaced%]
[%@total_under_5_displaced_scale%]
[%@total_under_5_displaced_percentage%]
[%@total_under_5_displaced_header%]
[%@total_over_60_displaced%]
[%@total_over_60_displaced_scale%]
[%@total_over_60_displaced_percentage%]
```

**Comment by myarjunar** [comment]
> We should be getting the diplacement etc numbers from the data table

@timlinux we still need a lot of vars if we use data table to get the value (number), because there are a lot of dynamic label that may changed based on the value from the table. For example when we want to show the data about displaced infant we need:
- denominated value of the total displaced infant. we could get this from the table.
- scale of the value (thousand, mill, bill, etc). we could store it either as a qgis variable or token and calculate the value on runtime.
- the percentage. we could store it either as a qgis variable or token and calculate the value on runtime.

![selection_118](https://user-images.githubusercontent.com/11134669/28769712-cea5fa76-7606-11e7-81ec-66b8a6938116.png)

**Comment by ivanbusthomi** [comment]
Hi mas @myarjunar - this is the template with variable

[infographic_variable.zip](https://github.com/inasafe/inasafe/files/1190201/infographic_variable.zip)

**Comment by myarjunar** [comment]
closed in #4353
